lib/web_ui/lib/src/engine/canvaskit/layer.dart

```diff
@@ -400,13 +400,14 @@ class ImageFilterEngineLayer extends ContainerLayer
 
   @override
   void preroll(PrerollContext prerollContext, Matrix4 matrix) {
-    final Matrix4 transform =
-        (_filter as CkManagedSkImageFilterConvertible).transform;
-    final Matrix4 childMatrix = matrix.multiplied(transform);
-    prerollContext.mutatorsStack.pushTransform(transform);
+    final Matrix4 childMatrix = Matrix4.copy(matrix);
+    childMatrix.translate(_offset.dx, _offset.dy);
+    prerollContext.mutatorsStack
+        .pushTransform(Matrix4.translationValues(_offset.dx, _offset.dy, 0.0));
```

Comment:
May need to add a golden test if this change doesn't trigger an existing one.

Reply:
Done. PTAL